### DISCUSSION TITLE
Add baseUrl variable to RequestHelpers templating

### DISCUSSION THREAD

**Discussion by louiealmeda** [mockoon/mockoon#466] [Ideas] [answered]
We are able to customize the response with the templating helpers described here https://mockoon.com/docs/latest/templating/mockoon-request-helpers/ 

Recently, I wanted to return an absolute path on a response, ideally it starts with the baseUrl where my current mockAPI is served on. The sample request response looks like this:

`GET: http://localhost:3001/api/users/123`
```
{
    "id": "123"
    "name": "Foobar",
    "imageUrl": "http://localhost:3001/api/images/123"
}
```

So templating for that looks like:
`GET: /users/:userId`
```
{
    "id": "{{urlParam 'userId'}}"
    "name": "Foobar",
    "imageUrl": "{{baseUrl}}/images/{{urlParam 'userId'}}"
}
```

I can't seem to achieve the same result with the existing helpers we have available. Would this be a good idea? or are there any other way to achieve this?

Thanks!

**Comment by 255kb** [comment, marked as answer]
Hi @louiealmeda there is currently no way to achieve this, we would need to add a `baseUrl` helper as you suggested which would return host, port and prefix. Could you please open an issue on this repository in order to track this feature? https://github.com/mockoon/commons-server
Thanks!

**Reply by louiealmeda** [reply]
Opened the issue here https://github.com/mockoon/commons-server/issues/7, Thanks!

**Reply by 255kb** [reply]
perfect, thanks!
